assert/equal_test.ts

```diff
@@ -276,3 +276,30 @@ Deno.test("equal() WeakMap, WeakRef and WeakSet", () => {
   assertFalse(equal(new WeakSet(), { constructor: WeakSet }));
   assertFalse(equal(new WeakRef({}), { constructor: WeakRef }));
 });
+
+Deno.test("equal() fast path for primitive keyed collections", () => {
+  const arr = Array.from({ length: 10_000 }, (_, i) => i);
+
+  function makeSet(arr: number[]) {
+    return new Set(arr);
+  }
+  function makeMap(arr: number[]) {
+    return new Map(arr.map((v, i) => [i, v]));
```

Comment:
```suggestion
    return new Map(arr.map((v) => [v, v]));
```

Reply:
That will mean the final key of the `with(-1, -1)` versions differs as well as the value. I guess probably best to include both cases.